ivy_tests/test_ivy/test_frontends/test_tensorflow/test_metrics.py

```diff
@@ -566,3 +566,37 @@ def test_tensorflow_metrics_mean_squared_logarithmic_error(
         y_true=np.asarray(x[0], dtype=input_dtype[0]),
         y_pred=np.asarray(x[1], dtype=input_dtype[1]),
     )
+
+
+@handle_cmd_line_args
+@given(
+    y_true_dtype_y_pred=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("numeric"),
+        min_value=1,
+        max_value=10,
+        shape=(2, 3),
+        shared_dtype=True,
+    ),
+    num_positional_args=helpers.num_positional_args(
+        fn_name="ivy.functional.frontends.tensorflow.cosine_similarity"
+    ),
+)
+def test_cosine_similarity(
+    y_true_dtype_y_pred, as_variable, num_positional_args, native_array, fw, with_out
+):
+    dtype, y_true_y_pred = y_true_dtype_y_pred
+    y_true_y_pred = ivy.array(y_true_y_pred, dtype=dtype)
+    y_true, y_pred = y_true_y_pred

```

Comment:
You are unstacking an array of dim [2,3] into two arrays of [1,3] each. What you should be doing instead is pass `num_arrays = 2`, and create np.arrays inside the function call to `test_frontend_function`, look at other functions in the same file. The current approach limits the dtype to only 1, which is throwing an error, since we generate one array only, however we are passing two arrays but one dtype. Hope that makes sense, looking forward to the update, thanks🙂

Reply:
Sure! I will get on it. I will request a review once I make the changes. I tried running tests locally and tbh `jaxlib` package is causing a lot of trouble on Windows. I am thinking to install an Ubuntu partition and test there. Windows is crap :\

Reply:
Hey, @Aarsh2001 I have changed the test function and it passed the test when I ran it locally (thanks to Linux :) ). Can you review the code and suggest the necessary changes?